### PR TITLE
Make TrustedState serializable, Fix #161

### LINKED ISSUE #161: No serialization support for `TrustedState`
Since `TrustedState` is supposed to be stored, so it'll need serialization. But users can't implement trait for external types, so it's better to builtin?

### PR DESCRIPTION
**Pull request by yihuang**
- Add Serialize/Deserialize instances to TrustedState and SignedHeader
- Change MemStore to use serialization

### CODE REVIEW

File: tendermint-lite/src/store.rs [hunk 1]
```diff
@@ -25,17 +25,18 @@ impl MemStore {
...
 
-    pub fn get(&self, h: Height) -> Result<&State, Error> {
+    pub fn get(&self, h: Height) -> Result<State, Error> {
         let mut height = h;
         if h == 0 {
             height = self.height
         }
         match self.store.get(&height) {
-            Some(state) => Ok(state),
+            Some(s) => Ok(serde_json::from_str(&s).unwrap()),
```
Comment (Liamsi) [inline tendermint-lite/src/store.rs]: What is the reason for serializing in the MemStore?

Comment (yihuang) [reply tendermint-lite/src/store.rs]: It's only to demostrate it's serializable, but I'm ok to remove it.

**Comment by yihuang** [comment]
> Without a good reason, I'd vote to not change the `MemStore`(I don't see why we'd want to store State as JSON in memory). Other than that this LGTM 👍

Done

**Comment by Liamsi** [comment]
ref #161

**Pull request merged**